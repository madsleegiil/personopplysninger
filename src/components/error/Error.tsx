import React from "react";
import AlertStripe from "nav-frontend-alertstriper";

export interface HTTPError {
  code: string;
  text: string;
}

interface props {
  error: HTTPError;
}

const Error = (props: props) => {
  const { error } = JSON.parse(JSON.stringify(props));

  return (
    <div className="error__container">
      <AlertStripe type="advarsel" solid>
        Oisann, noe gikk galt!
        <span>{` ${error.code}: ${error.text}`}</span>
      </AlertStripe>
    </div>
  );
};

export default Error;
