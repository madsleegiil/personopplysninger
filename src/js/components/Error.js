import React from "react";
import PropTypes from "prop-types";
import AlertStripe from "nav-frontend-alertstriper";

const Error = props => {
  const { statusCode } = props;
  return (
    <div className="BoxContainer">
      <AlertStripe type="advarsel" solid>
        Oisann, noe gikk galt!
        {statusCode === 500 ? (
          <span> 500 Internal Server Error</span>
        ) : (
          statusCode
        )}
      </AlertStripe>
    </div>
  );
};

Error.propTypes = {
  statusCode: PropTypes.number
};

Error.defaultProps = {
  statusCode: null
};

export default Error;
