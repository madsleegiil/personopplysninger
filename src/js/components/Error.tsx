import React from "react";
import PropTypes from "prop-types";
import AlertStripe from "nav-frontend-alertstriper";

const Error = (props: any) => {
  const { error } = props;
  return (
    <div className="BoxContainer">
      <AlertStripe type="advarsel" solid>
        Oisann, noe gikk galt!
        <span>{` ${error.code}: ${error.text}`}</span>
      </AlertStripe>
    </div>
  );
};

Error.propTypes = {
  error: PropTypes.shape({
    code: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired
  }).isRequired
};

export default Error;
