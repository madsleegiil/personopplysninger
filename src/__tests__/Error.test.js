import * as React from 'react';
import Error from 'js/components/Error';
import wrapIntl from 'js/IntlTestHelper';
const ReactTestRenderer = require('react-test-renderer');

test('render Error', () => {
  const component = ReactTestRenderer.create(wrapIntl(<Error
    statusCode={500}
  />));
  expect(component.toJSON()).toMatchSnapshot();
});
