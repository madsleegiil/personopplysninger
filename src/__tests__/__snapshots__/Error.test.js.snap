// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Error 1`] = `
<div
  className="BoxContainer"
>
  <div
    aria-live="assertive"
    className="alertstripe alertstripe--solid alertstripe--advarsel"
  >
    <span
      aria-label="advarsel"
      className="alertstripe__ikon"
    >
      <svg
        focusable="false"
        height="1.5em"
        kind="feil-sirkel-fyll"
        viewBox="0 0 24 24"
        width="1.5em"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M11.999 0C5.395 0 .013 5.372 0 11.976a11.923 11.923 0 0 0 3.498 8.493A11.925 11.925 0 0 0 11.977 24H12c6.603 0 11.986-5.373 12-11.978C24.013 5.406 18.64.012 11.999 0z"
            fill="#A13A28"
            fillRule="nonzero"
          />
          <path
            d="M12 10.651l3.372-3.372a.954.954 0 1 1 1.349 1.35L13.349 12l3.372 3.372a.954.954 0 1 1-1.35 1.349L12 13.349 8.628 16.72a.954.954 0 1 1-1.349-1.35L10.651 12 7.28 8.628A.954.954 0 1 1 8.63 7.28L12 10.651z"
            fill="#FFF"
            fillRule="nonzero"
          />
        </g>
      </svg>
    </span>
    <span
      className="typo-normal alertstripe__tekst"
    >
      Oisann, noe gikk galt! 
      <span>
        500 Internal Server Error
      </span>
    </span>
  </div>
</div>
`;
